### PR TITLE
Introduce action metrics for mv3 service worker restart

### LINKED ISSUE #17197: Action Queue metrics
We only need one event: tracking when the action queue was utilized upon a service worker restart. Each event should record the names of the methods that were processed by the action queue.

### LINKED ISSUE #17198: Service Worker metrics

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## Explanation

Introduce action metrics for logging:
- List of actions processed during a service worker restart (action queue processing)
- Time between service worker restarts.

This PR does not include any tests.

* Fixes #17197 (apart from tests)
* Fixes: #17198 (apart from tests)

## Automated Testing Steps

1. Copy the test on this PR https://github.com/MetaMask/metamask-extension/pull/18347/files
2. Run `yarn build:test:dev:mv3`
3. On a separate window, run `IN_TEST=true yarn test:e2e:single test/e2e/mv3/service-worker-restart.spec.js --browser=chrome --leave-running`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [ ] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: app/manifest/v3/chrome.json [hunk 1]
```diff
@@ -6,19 +6,5 @@
     "matches": ["https://metamask.io/*"],
     "ids": ["*"]
   },
-  "minimum_chrome_version": "80",
-  "permissions": [
-    "storage",
-    "unlimitedStorage",
-    "clipboardWrite",
-    "http://localhost:8545/",
```
Comment (PeterYinusa) [inline app/manifest/v3/chrome.json:14]: It looks like some of the permissions have been removed, do we no longer need them?

File: package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
     "build:test": "SEGMENT_HOST='https://api.segment.io' SEGMENT_WRITE_KEY='FAKE' SENTRY_DSN_DEV=https://fake@sentry.io/0000000 PORTFOLIO_URL=http://127.0.0.1:8080 yarn build test",
     "build:test:flask": "yarn build test --build-type flask",
     "build:test:mv3": "ENABLE_MV3=true SEGMENT_HOST='https://api.segment.io' SEGMENT_WRITE_KEY='FAKE' SENTRY_DSN_DEV=https://fake@sentry.io/0000000 PORTFOLIO_URL=http://127.0.0.1:8080 yarn build test",
+    "build:test:dev:mv3": "IN_TEST=true ENABLE_MV3=true SEGMENT_HOST='https://api.segment.io' SEGMENT_WRITE_KEY='FAKE' SENTRY_DSN_DEV=https://fake@sentry.io/0000000 PORTFOLIO_URL=http://127.0.0.1:8080 yarn build:dev testDev --apply-lavamoat=false",
```
Comment (PeterYinusa) [inline package.json:24]: Do we need `IN_TEST=true` inline here? I think its set automatically based on the `testDev` build target provided in this same command

File: package.json [hunk 3]
```diff
@@ -32,6 +33,8 @@
     "test:unit:mocha": "node ./test/run-unit-tests.js --mocha",
     "test:e2e:chrome": "SELENIUM_BROWSER=chrome node test/e2e/run-all.js",
     "test:e2e:chrome:snaps": "SELENIUM_BROWSER=chrome node test/e2e/run-all.js --snaps",
+    "test:e2e:chrome:nft": "SELENIUM_BROWSER=chrome node test/e2e/run-all.js --nft",
```
Comment (PeterYinusa) [inline package.json:36]: This script has been removed from develop as the NFT tests are part of the mail build now. Is there a reason we need to reintroduce it here?

File: test/e2e/webdriver/driver.js [hunk 4]
```diff
@@ -482,20 +478,22 @@ class Driver {
...
-        const eventDescription = event.args.filter(
+        const eventDescriptions = event.args.filter(
           (err) => err.description !== undefined,
         );
-        const [{ description }] = eventDescription;
+
+        const [eventDescription] = eventDescriptions;
         const ignore = ignoredErrorMessages.some((message) =>
-          description.includes(message),
+          eventDescription?.description.includes(message),
```
Comment (PeterYinusa) [inline test/e2e/webdriver/driver.js:492]: 👍

File: app/manifest/v3/chrome.json (hunk 1 above)
Comment (PeterYinusa) [reply app/manifest/v3/chrome.json:14]: Thanks for the explanation, so does that mean we consume the v2 base for the v3 files?

File: ui/components/ui/box/box.js [hunk 5]
```diff
@@ -300,8 +301,14 @@ const Box = React.forwardRef(function Box(
...
   }
   const Component = as;
+
   return (
-    <Component className={boxClassName} ref={ref} {...props}>
+    <Component
+      className={boxClassName}
+      ref={ref}
+      aria-label={ariaLabel}
+      {...{ ...props, ariaLabel }}
```
Comment (brad-decker) [inline ui/components/ui/box/box.js:310]: Is this related to action metrics?

**Review by brad-decker — DISMISSED** [review]
Just a question

File: ui/components/ui/box/box.js (hunk 5 above)
Comment (pedronfigueiredo) [reply ui/components/ui/box/box.js]: great comment Brad, this has now been addressed on [another PR](https://github.com/MetaMask/metamask-extension/pull/18370), and I am removing the changes from here

**Review by brad-decker — DISMISSED** [review]

File: app/manifest/v3/chrome.json (hunk 1 above)
Comment (pedronfigueiredo) [reply app/manifest/v3/chrome.json:14]: I don't think so. I think v3 just has had a different set of permissions from v2, notably, we don't seem to need to add URLs to the permissions file.

For a more detailed view of the original state of this file, the changes that caused issues see here:

https://github.com/MetaMask/metamask-extension/pull/18218/files

This PR has now been reverted atomically here https://github.com/MetaMask/metamask-extension/pull/18336/files

So I'll remove the changes from this PR

Comment (PeterYinusa) [reply app/manifest/v3/chrome.json:14]: > we don't seem to need to add URLs to the permissions file.

Thanks that clears up the confusion for me.

**Review by brad-decker — DISMISSED** [review]

**Review by danjm — DISMISSED** [review]
LGTM

**Review by PeterYinusa — DISMISSED** [review]

**Review by danjm — APPROVED** [review]

**Review by PeterYinusa — APPROVED** [review]
